### PR TITLE
Fix *ModelFactory namespaces

### PR DESCRIPTION
**Pull request by AlexanderSher**

### CODE REVIEW

File: sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridModelFactory.cs [hunk 1]
```diff
@@ -5,11 +5,13 @@
 using System.Collections.Generic;
 using System.ComponentModel;
 using System.Text.Json;
+using Azure.Core;
 using Azure.Messaging.EventGrid.SystemEvents;
 
 namespace Azure.Messaging.EventGrid
 {
 #pragma warning disable CA1054 // URI-like parameters should not be strings
+    [CodeGenType(nameof(EventGridModelFactory))]
```
Comment (JoshLove-msft) [inline sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridModelFactory.cs:14]: What is the impact of this change?

Comment (AlexanderSher) [reply sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridModelFactory.cs:14]: This prevents `EventGridModelFactory` from being generated in `Models` namespace cause `EventGridModelFactory` has been GA'd

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@
 using System.Linq;
 using Azure.AI.TextAnalytics.Models;
 
-namespace Azure.AI.TextAnalytics
+namespace Azure.AI.TextAnalytics.Models
```
[the remark this answers is not in the document]
Comment (AlexanderSher) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs:10]: Ok. Btw, `TextAnalyticsWarningInternal` is in the `Models` namespace. Do you want me to fix that as well?

[the remark this answers is not in the document]
Comment (AlexanderSher) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs]: Fixed

**Review by maririos — APPROVED** [review]
TA and Translation LGTM

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**